lib/spack/spack/graph.py

```diff
@@ -42,54 +42,84 @@
 can take a number of specs as input.
 
 """
+import heapq
+import itertools
 import sys
-from heapq import heapify, heappop, heappush
 
-from llnl.util.tty.color import ColorStream
+import llnl.util.tty.color
 
-from spack.dependency import all_deptypes, canonical_deptype
+import spack.dependency
```

Comment:
```python
import spack.dependency as dp
```

would make this a bit more readable

Reply:
Hmm, not sure. Using the fully qualified name I know from where a name is pulled without looking at a context that might be hundreds of lines above.